utils/availability-macros.def

```diff
@@ -24,18 +24,18 @@ SwiftStdlib 9999:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999
 # development. This usually resolves to placeholder version numbers (9999) until
 # the corresponding operating systems get announced.
 
-SwiftStdlib 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2
-SwiftStdlib 5.1:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0
-SwiftStdlib 5.2:macOS 10.15.4, iOS 13.4, watchOS 6.2, tvOS 13.4
-SwiftStdlib 5.3:macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0
-SwiftStdlib 5.4:macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5
-SwiftStdlib 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0
-SwiftStdlib 5.6:macOS 12.3, iOS 15.4, watchOS 8.5, tvOS 15.4
-SwiftStdlib 5.7:macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0
-SwiftStdlib 5.8:macOS 13.3, iOS 16.4, watchOS 9.4, tvOS 16.4
-SwiftStdlib 5.9:macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0
-SwiftStdlib 5.10:macOS 14.4, iOS 17.4, watchOS 10.4, tvOS 17.4
-SwiftStdlib 6.0:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999
+SwiftStdlib 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2, visionOS 1.0
+SwiftStdlib 5.1:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, visionOS 1.0
+SwiftStdlib 5.2:macOS 10.15.4, iOS 13.4, watchOS 6.2, tvOS 13.4, visionOS 1.0
+SwiftStdlib 5.3:macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0, visionOS 1.0
+SwiftStdlib 5.4:macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5, visionOS 1.0
+SwiftStdlib 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, visionOS 1.0
+SwiftStdlib 5.6:macOS 12.3, iOS 15.4, watchOS 8.5, tvOS 15.4, visionOS 1.0
+SwiftStdlib 5.7:macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, visionOS 1.0
+SwiftStdlib 5.8:macOS 13.3, iOS 16.4, watchOS 9.4, tvOS 16.4, visionOS 1.0
+SwiftStdlib 5.9:macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, visionOS 1.0
```

Comment:
There usually isn't much point to listing the first release of a platform in availability lists -- historically we just omitted the version incantations if the API was always available. Do we really need `visionOS 1.0` for these?

Reply:
Artem and I discussed this and while what you're saying is true in general, the fact that the compiler has to re-map iOS availability to visionOS availability means makes this less straightforward than the normal behavior of availability when an explicit annotation is for a platform is missing. That said, I can't remember exactly what the problem was when we omitted these explicit versions so we should look at it again.

Reply:
I did some experiments and ended up removing the `..., visionOS 1.0` here, and then therefore also reverting various places in the stdlib where I had to replace:
```
@backDeployed(before: SwiftStdlib 5.9)
```
with:
```
// Corresponds to SwiftStdlib 5.9 without visionOS
@backDeployed(before: macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0)
```
Looking back on it, I think this was a WIP solution to a problem that we seem to have ended up resolving anyways, as this change did not affect our test suite results. Thanks @lorentey.